### PR TITLE
tests: regression tests for types sent over wire as json 

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
## Description

Adds regression tests for the types that are sent over the wire with IBC as JSON bytes. These tests use a hardcoded expected JSON schema that should never change lest we break the wire. 

Note I've cherry-picked these tests into `release/v1.0.x` and they pass as expected, woot we haven't broken the wire

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

File: x/ccv/consumer/keeper/relay_test.go [hunk 1]
```diff
@@ -234,13 +234,21 @@ func TestOnAcknowledgementPacket(t *testing.T) {
...
+	consumerPacketData := types.NewConsumerPacketData(
+		types.SlashPacket,
+		&types.ConsumerPacketData_SlashPacketData{
+			SlashPacketData: slashPacketData,
+		},
+	)
+
 	// AcknowledgePacket is in reference to a packet originally sent from this (consumer) module.
 	packet := channeltypes.NewPacket(
-		packetData.GetBytes(),
```
Comment (smarshall-spitzbart) [inline x/ccv/consumer/keeper/relay_test.go:243]: Wrong type was previously used in this test. I've adjusted it to use the proper consumer packet data type

File: x/ccv/types/ccv.go [hunk 2]
```diff
@@ -48,11 +50,6 @@ func (mat VSCMaturedPacketData) ValidateBasic() error {
 	return nil
 }
 
-func (mat VSCMaturedPacketData) GetBytes() []byte {
```
Comment (smarshall-spitzbart) [inline x/ccv/types/ccv.go:51]: This and the other GetBytes methods should never be used. VSCMatured and SlashPacket data are both wrapped as ConsumerPacketData when sent over the wire. The ConsumerPacketData.GetBytes() method should be used and is tested in this PR

**Review by MSalopek — APPROVED** [review]
Thanks!

LGTM!

**Review by mpoke — APPROVED** [review]
LGTM

**Pull request merged**